frontend/src/model/beta/images/CodeLine.tsx

```diff
@@ -1,35 +1,44 @@
-import { Box, Tooltip } from '@mui/material'
+import CodeIcon from '@mui/icons-material/Code'
+import { Box, Stack, Tooltip, Typography } from '@mui/material'
 import { useTheme } from '@mui/material/styles'
+import { ReactElement } from 'react'
 import useNotification from 'src/common/Snackbar'
 
-export default function CodeLine({ line }: { line: string }) {
+interface CodeLineProps {
+  line: string
+  icon?: ReactElement
+}
+
+export default function CodeLine({ line, icon = <CodeIcon /> }: CodeLineProps) {
   const theme = useTheme()
   const sendNotification = useNotification()
 
-  const handleButtonClick = () => {
+  function handleButtonClick() {
     navigator.clipboard.writeText(line)
     sendNotification({ variant: 'success', msg: 'Copied to clipboard' })
   }
 
   return (
-    <div
-      style={{
-        cursor: 'pointer',
-      }}
-      role='button'
-      tabIndex={0}
-      onClick={() => {
-        handleButtonClick()
-      }}
-      onKeyDown={(e) => {
-        if (e.key === 'Enter') {
+    <Tooltip title='Copy to clipboard' arrow>
+      <Box
+        sx={{ backgroundColor: theme.palette.container.main, p: 1, borderRadius: 2, cursor: 'pointer' }}
+        component='div'
+        role='button'
+        tabIndex={0}
+        onClick={() => {
           handleButtonClick()
-        }
-      }}
-    >
-      <Tooltip title='Copy to clipboard' arrow>
-        <Box sx={{ backgroundColor: theme.palette.container.main, p: 1, borderRadius: 2 }}>$ {line}</Box>
-      </Tooltip>
-    </div>
+        }}
+        onKeyDown={(e) => {
+          if (e.key === 'Enter' || e.key === 'Spacebar') {
```

Comment:
For all modern browsers the key of spacebar will be `" "`. See: https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values#whitespace_keys

If we want to support older versions of Firefox (pre version 37) then `'Spacebar'` will also be required.